### PR TITLE
TALC compatibility changes

### PR DESCRIPTION
**Pull request by kasra-keshavarz**
Necessary changes for the workflows/assignments of this course. Various commits describe the necessary changes, but in summary:
* A new list of modules need to be loaded every time users need to run the workflows in terminal,
* Necessary changes has been implemented in the relevant Jupyter Notebooks,
* Adding instruction for activating the Python environment for this course,
* Adding necessary instruction in the Jupyter Notebook,
* SUMMA has been re-compiled on TALC, so the relevant paths have been changes,
* The workflow to compile SUMMA is not part of this PR,
* New instructions will follow based on needs

### CODE REVIEW

File: 0_prerequisits/hpc-env/README.md
Comment (wknoben) [inline 0_prerequisits/hpc-env/README.md:1]: Minor, but `prerequisites` needs an extra `e`

File: 0_prerequisits/hpc-env/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Saving list of modules for `TALC`
+
+To save the list of modules, copy the [saved list of modules](./pbhm-mods) to the `$HOME/.module` directory.
+```console
+$ mkdir $HOME/.module
+$ cp pbhm-mods $HOME/.module
```
Comment (wknoben) [inline 0_prerequisits/hpc-env/README.md]: Would be good to specify if students need to `cd` into a specific dir before executing this command:
```
cd $HOME/pbhmCourse_student/0_prerequisites/hpc-env
```

Comment (wknoben) [reply 0_prerequisits/hpc-env/README.md]: Changing this line to 
```
cp /work/TALC/enci619.05_2024w/local/modules/pbhm-mods $HOME/.module
```
Would maybe also fix this?

File: 0_prerequisits/hpc-env/README.md
Comment (wknoben) [reply 0_prerequisits/hpc-env/README.md:1]: affects copy command too

File: 0_prerequisits/jupyter/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# TALC Jupyter enviornment
+The Jupyter environment for the `TALC` cluster is located in the following: https://talc.ucalgary.ca:8000/hub/login. You can use your UC username and password to login.
```
Comment (wknoben) [inline 0_prerequisits/jupyter/README.md]: Might be good to add a specific call out that in this case, user name does not mean email address. E.g., use only the part before the `@` in `someone@ucalgary.ca`.
I know the instructions already say user name (not email), but in almost every other SSO UCal uses email address is either needed or also works - this one seems an exception to that

File: 0_prerequisits/python-env/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Using the dedicated Python enviornment in terminal
```
Comment (wknoben) [inline 0_prerequisits/python-env/README.md]: I expect we can cover this in the assignment description, but might be good to specify that this takes place in the terminal after ssh'ing into TALC

File: 0_prerequisits/jupyter/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# TALC Jupyter enviornment
```
Comment (wknoben) [inline 0_prerequisits/jupyter/README.md]: I expect we can cover this in the assignment description, but might be good to specify that this takes place in a standard web browser, after the other steps have been completed

File: 0_prerequisits/hpc-env/README.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# Saving list of modules for `TALC`
```
Comment (wknoben) [inline 0_prerequisits/hpc-env/README.md]: I expect we can cover this in the assignment description, but might be good to specify that this takes place in the terminal after ssh'ing into TALC

File: 0_prerequisits/python-env/README.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+$ source /work/TALC/enci619.05_2024w/local/python/pbhm-python-venv/bin/activate
+$ python -m ipykernel install --user --name "pbhm-python-env"
+```
+After this one-time step, you can use this environment as a dedicated Kernel on your Jupyter Notebooks.
+
+> [!NOTE]
+> You only need to install the Python enviornment once to use in Jupyter Notebooks.
+
+> [!CAUTION]
+> You need to select the appropriate Python environment as a Kernel to run Notebooks for Assignment #3. After installing the Python environment as a Kernel, you can find the Kernel name in Jupyter Nobteooks under the `Kernel > Change kernel > pbhm-python-env` menu.
```
Comment (wknoben) [inline 0_prerequisits/python-env/README.md]: In my case it popped a warning/error when opening the first notebook. Might be good to mention that possibility as well

File: 3_process_based_modelling/1a_pysumma_functionality.ipynb
Comment (wknoben) [inline 3_process_based_modelling/1a_pysumma_functionality.ipynb:1]: The plotting is incredibly slow (over VPN at least), but this completes as expected

File: 3_process_based_modelling/1b_pysumma_example.ipynb
Comment (wknoben) [inline 3_process_based_modelling/1b_pysumma_example.ipynb:1]: Completes as expected

File: 3_process_based_modelling/2a_reynolds_mountain_east.ipynb
Comment (wknoben) [inline 3_process_based_modelling/2a_reynolds_mountain_east.ipynb:1]: Completes as expected - tested with full answers to all exercises in this notebook

Comment (wknoben) [inline 3_process_based_modelling/2a_reynolds_mountain_east.ipynb:1]: Would be good to clear all cell outputs before merging - this keeps the history smaller

File: 3_process_based_modelling/2b_plumber_sites.ipynb
Comment (wknoben) [inline 3_process_based_modelling/2b_plumber_sites.ipynb:1]: Would be good to clear all cell outputs before merging

File: 3_process_based_modelling/2c_bow_river_above_banff.ipynb
Comment (wknoben) [inline 3_process_based_modelling/2c_bow_river_above_banff.ipynb:1]: This file needs updates to the SUMMA exe location and library locations

Comment (wknoben) [inline 3_process_based_modelling/2c_bow_river_above_banff.ipynb:1]: Completes as expected - tested with downloading hydroshare resource and answers to all questions

File: 4_high_performance_computing/scripts/ex_2.1_merge_output.sh
Comment (wknoben) [inline 4_high_performance_computing/scripts/ex_2.1_merge_output.sh:1]: libraries load as expected

File: 3_process_based_modelling/2b_plumber_sites.ipynb
Comment (wknoben) [inline 3_process_based_modelling/2b_plumber_sites.ipynb:1]: Still running for the sake of it, but so far it seems to work

**Review by wknoben — CHANGES_REQUESTED** [review]

File: 0_prerequisits/hpc-env/README.md (hunk 1 above)
Comment (kasra-keshavarz) [reply 0_prerequisits/hpc-env/README.md]: So, the file is also available where this README.md file is located in this repository. But, I've changed it to the line you've mentioned here.

File: 0_prerequisits/python-env/README.md (hunk 6 above)
Comment (kasra-keshavarz) [reply 0_prerequisits/python-env/README.md]: What was the warning/error? This should not happen.

File: 3_process_based_modelling/1a_pysumma_functionality.ipynb
Comment (kasra-keshavarz) [reply 3_process_based_modelling/1a_pysumma_functionality.ipynb:1]: Yeah, sorry about this. There isn't much we could do at our end.

File: 3_process_based_modelling/1b_pysumma_example.ipynb
Comment (kasra-keshavarz) [reply 3_process_based_modelling/1b_pysumma_example.ipynb:1]: Great

File: 3_process_based_modelling/2a_reynolds_mountain_east.ipynb
Comment (kasra-keshavarz) [reply 3_process_based_modelling/2a_reynolds_mountain_east.ipynb:1]: Fantastic

File: 3_process_based_modelling/2c_bow_river_above_banff.ipynb
Comment (kasra-keshavarz) [reply 3_process_based_modelling/2c_bow_river_above_banff.ipynb:1]: Done, thanks.

Comment (kasra-keshavarz) [reply 3_process_based_modelling/2c_bow_river_above_banff.ipynb:1]: Perfect, thanks.

File: 4_high_performance_computing/scripts/ex_2.1_merge_output.sh
Comment (kasra-keshavarz) [reply 4_high_performance_computing/scripts/ex_2.1_merge_output.sh:1]: Great

File: 3_process_based_modelling/2b_plumber_sites.ipynb
Comment (kasra-keshavarz) [reply 3_process_based_modelling/2b_plumber_sites.ipynb:1]: Yep, I remember this taking a while to run. But, it ran on my end without any issues.

**Comment by kasra-keshavarz** [comment]
I've tried to address your comments, let me know what you think.

File: 0_prerequisits/python-env/README.md (hunk 6 above)
Comment (kasra-keshavarz) [reply 0_prerequisits/python-env/README.md]: Make a note, so students don't freak out.

Comment (kasra-keshavarz) [reply 0_prerequisits/python-env/README.md]: Done as well.

**Review by wknoben — APPROVED** [review]
looks good to me

**Pull request merged**